packages/completer/src/handler.ts

```diff
@@ -376,6 +417,64 @@ export class CompletionHandler implements IDisposable {
       });
   }
 
+  private async _makeInlineRequest(
+    position: CodeEditor.IPosition,
+    trigger: InlineCompletionTriggerKind
+  ) {
+    const editor = this.editor;
+
+    if (!editor) {
+      return Promise.reject(new Error('No active editor'));
+    }
+    if (!this.inlineCompleter) {
+      return Promise.reject(new Error('No inline completer'));
+    }
+
+    const line = editor.getLine(position.line);
+    if (typeof line === 'undefined' || position.column < line.length) {
+      // only auto-trigger on end of line
+      return;
+    }
+
+    const request = this._composeRequest(editor, position);
+
+    const model = this.inlineCompleter.model;
+    if (!model) {
+      return;
+    }
+    model.cursor = position;
+
+    const current = ++this._fetchingInline;
+    const promises = this._reconciliator.fetchInline(request, trigger);
+
+    let i = 0;
+    for (const promise of promises) {
```

Comment:
following on the linter comment

I guess it will be better to do something like

```ts
await Promise.all(promises.map(async p => {
  const result = await p;
  /* ... */
}));

Reply:
I had addressed the linter comment in d93acdb22b9742b1f409d45bf8adf7296303f5ce. I feel that keeping hanging promise here makes it clear that we terminate the promise chain here and the rest happens in callbacks.